python/ray/util/client/server/proxier.py

```diff
@@ -150,18 +151,22 @@ def _get_redis_address(self) -> str:
         self._session_dir = connection_tuple["session_dir"]
         return self.redis_address
 
-    def _get_session_dir(self) -> str:
+    def _get_node(self) -> Any:
         """
         Gets the session_dir of this running Ray session. This usually
         looks like /tmp/ray/session_<timestamp>.
         """
-        if self._session_dir:
-            return self._session_dir
-        # Connect a driver to an already running cluster.
-        connection_tuple = ray.init(address=self._get_redis_address())
-        ray.shutdown()
-        self._session_dir = connection_tuple["session_dir"]
-        return self._session_dir
+        if self._node:
+            return self._node
+
+        self._node = ray.node.Node(
+            RayParams(redis_address=self._get_redis_address()),
+            head=False,
+            shutdown_at_exit=False,
+            spawn_reaper=False,
+            connect_only=True)
+
+        return self._node
```

Comment:
why do we need self._node if we return it?

Reply:
looks like to get the session_dir in create_specific_server below

Reply:
I'm using the 'getter' pattern here. I should explicitly use an `@getter` decorated method instead. 

The reason is that I want to avoid calling 
```
if self._node is None:
    ... self._create_node
```
every time I want to access self._node